compiler/expressions.go

```diff
@@ -1112,6 +1112,12 @@ func (c *funcContext) translateImplicitConversion(expr ast.Expr, desiredType typ
 	}
 
 	exprType := c.p.TypeOf(expr)
+
+	// whenever expr has an underlying *types.Interface type we need to $copyInterfaceVal
+	if _, isIface := exprType.Underlying().(*types.Interface); isIface {
```

Comment:
Cool. Just a small remark:

> I wasn't sure whether I'd considered all cases

Imho don't let that bother you. In my experience it is better to aim for the simplest code that handles the cases you know.